src/js/stores/AppsStore.js

```diff
@@ -146,12 +146,29 @@ function processApp(app) {
   return app;
 }
 
-function processApps(apps) {
-  return apps.map(function (app) {
-    return processApp(app);
+function processEmptyGroup(group) {
+  return Object.assign(processApp({id: group.id}), {
+    isGroup: true,
+    status: null
   });
 }
 
+function processAppsAndGroups(group, apps = []) {
```

Comment:
Yes, I think it is the right time now to switch over to the groups-endpoint completely.
Due to the nested (non-flat) structure returned by this endpoint, it is obvious which entry is a group and which one is an app, so this data doesn't need to be extended by ourselves.

@pierlo-upitup I agree with all your 5 points you suggested here for the refactoring. They sound really reasonable.
And I like the idea to flatten the apps out of the groups structure with the apps-getter as you suggested for the search results.
We should fully rely on the groups endpoint now.

More then 5 lines.. ;)